src/System.Linq/src/System/Linq/First.cs

```diff
@@ -10,47 +10,53 @@ public static partial class Enumerable
     {
         public static TSource First<TSource>(this IEnumerable<TSource> source)
         {
+            bool found;
+            TSource first = source.TryGetFirst(out found);
+            
+            if (!found)
+            {
+                throw Error.NoElements();
+            }
+
+            return first;
```

Comment:
I tried writing this as `return found ? first : throw Error.NoElements()` initially. But it looks like that feature of C# 7 isn't enabled yet :sad: